rust-runtime/aws-smithy-client/src/test_connection.rs

```diff
@@ -271,6 +273,62 @@ where
     }
 }
 
+/// Create a DynConnector from `Fn(http:Request) -> http::Response`
+///
+/// # Examples
+///
+/// ```rust
+/// use aws_smithy_client::test_connection::infallible_connection_fn;
+/// let connector = infallible_connection_fn(|_req|http::Response::builder().status(200).body("OK!").unwrap());
+/// ```
+pub fn infallible_connection_fn<B>(
+    f: impl Fn(http::Request<SdkBody>) -> http::Response<B> + Send + Sync + 'static,
+) -> DynConnector
+where
+    B: Into<SdkBody>,
+{
+    ConnectionFn::infallible(f)
+}
+
+#[derive(Clone)]
+struct ConnectionFn {
+    response: Arc<
```

Comment:
To satisfy CI with the new MSRV:
```suggestion
    #[allow(clippy::type-complexity)]
    response: Arc<
```